plugins/obs-ffmpeg/obs-ffmpeg-srt.h

```diff
@@ -0,0 +1,836 @@
+	    (s->ffs >= 0 && libsrt_setsockopt(h, fd, SRTO_FC, "SRTO_FC",
+					      &s->ffs, sizeof(s->ffs)) < 0) ||
+	    (s->ipttl >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_IPTTL, "SRTO_IPTTL", &s->ipttl,
+			       sizeof(s->ipttl)) < 0) ||
+	    (s->iptos >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_IPTOS, "SRTO_IPTOS", &s->iptos,
+			       sizeof(s->iptos)) < 0) ||
+	    (s->latency >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_LATENCY, "SRTO_LATENCY", &latency,
+			       sizeof(latency)) < 0) ||
+	    (s->rcvlatency >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_RCVLATENCY, "SRTO_RCVLATENCY",
+			       &rcvlatency, sizeof(rcvlatency)) < 0) ||
+	    (s->peerlatency >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_PEERLATENCY, "SRTO_PEERLATENCY",
+			       &peerlatency, sizeof(peerlatency)) < 0) ||
+	    (s->tlpktdrop >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_TLPKTDROP, "SRTO_TLPKTDROP",
+			       &s->tlpktdrop, sizeof(s->tlpktdrop)) < 0) ||
+	    (s->nakreport >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_NAKREPORT, "SRTO_NAKREPORT",
+			       &s->nakreport, sizeof(s->nakreport)) < 0) ||
+	    (connect_timeout >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_CONNTIMEO, "SRTO_CONNTIMEO",
+			       &connect_timeout,
+			       sizeof(connect_timeout)) < 0) ||
+	    (s->sndbuf >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_SNDBUF, "SRTO_SNDBUF", &s->sndbuf,
+			       sizeof(s->sndbuf)) < 0) ||
+	    (s->rcvbuf >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_RCVBUF, "SRTO_RCVBUF", &s->rcvbuf,
+			       sizeof(s->rcvbuf)) < 0) ||
+	    (s->lossmaxttl >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_LOSSMAXTTL, "SRTO_LOSSMAXTTL",
+			       &s->lossmaxttl, sizeof(s->lossmaxttl)) < 0) ||
+	    (s->minversion >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_MINVERSION, "SRTO_MINVERSION",
+			       &s->minversion, sizeof(s->minversion)) < 0) ||
+	    (s->streamid &&
+	     libsrt_setsockopt(h, fd, SRTO_STREAMID, "SRTO_STREAMID",
+			       s->streamid, (int)strlen(s->streamid)) < 0) ||
+#if SRT_VERSION_VALUE >= 0x010401
+	    (s->smoother &&
+	     libsrt_setsockopt(h, fd, SRTO_CONGESTION, "SRTO_CONGESTION",
+			       s->smoother, (int)strlen(s->smoother)) < 0) ||
+#else
+	    (s->smoother &&
+	     libsrt_setsockopt(h, fd, SRTO_SMOOTHER, "SRTO_SMOOTHER",
+			       s->smoother, (int)strlen(s->smoother)) < 0) ||
+#endif
+	    (s->messageapi >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_MESSAGEAPI, "SRTO_MESSAGEAPI",
+			       &s->messageapi, sizeof(s->messageapi)) < 0) ||
+	    (s->payload_size >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_PAYLOADSIZE, "SRTO_PAYLOADSIZE",
+			       &s->payload_size,
+			       sizeof(s->payload_size)) < 0) ||
+	    (/*(h->flags & AVIO_FLAG_WRITE) &&*/
+	     libsrt_setsockopt(h, fd, SRTO_SENDER, "SRTO_SENDER", &yes,
+			       sizeof(yes)) < 0) ||
+	    (s->tsbpd >= 0 &&
+	     libsrt_setsockopt(h, fd, SRTO_TSBPDMODE, "SRTO_TSBPDMODE",
+			       &s->tsbpd, sizeof(s->tsbpd)) < 0)) {
+		return AVERROR(EIO);
+	}
```

Comment:
This feels like it should be a macro or helper function, way too much code for a single statement.

Reply:
the conditions for the options are different; some must be >=0 , >=-1, or just exist. Not sure how to go about a macro. The code is strictly that in avformat/libsrt.c . I didn't change anything and maybe for the initial merge it's better to stay close to the source.